internal/cmp/lb_pools.go

```diff
@@ -0,0 +1,122 @@
+// (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+
+package cmp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/client"
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/models"
+	"github.com/HewlettPackard/hpegl-vmaas-terraform-resources/internal/utils"
+	"github.com/tshihad/tftags"
+)
+
+type loadBalancerPool struct {
+	lbClient *client.LoadBalancerAPIService
+}
+
+func newLoadBalancerPool(loadBalancerClient *client.LoadBalancerAPIService) *loadBalancerPool {
+	return &loadBalancerPool{
+		lbClient: loadBalancerClient,
+	}
+}
+
+func (lb *loadBalancerPool) Read(ctx context.Context, d *utils.Data, meta interface{}) error {
+	var lbPoolResp models.GetSpecificLBPoolResp
+	if err := tftags.Get(d, &lbPoolResp); err != nil {
+		return err
+	}
+
+	lbDetails, err := lb.lbClient.GetLoadBalancers(ctx)
+	if err != nil {
+		return err
+	}
+
+	getlbPoolResp, err := lb.lbClient.GetSpecificLBPool(ctx, lbDetails.GetNetworkLoadBalancerResp[0].ID, lbPoolResp.ID)
```

Comment:
lbDetails.GetNetworkLoadBalancerResp[0].ID, we cant assume this, please get this from user

Reply:
done